code/controllers/subsystem/shuttle.dm

```diff
@@ -659,14 +659,19 @@ SUBSYSTEM_DEF(shuttle)
 
 /datum/controller/subsystem/shuttle/proc/is_in_shuttle_bounds(atom/A)
 	var/area/current = get_area(A)
-	if(istype(current, /area/shuttle) && !istype(current, /area/shuttle/transit))
+	if((istype(current, /area/shuttle) && !istype(current, /area/shuttle/transit)) || istype(current, /area/ship))
```

Comment:
this file can be made modular if you make a subtype of it that isn't actually a subtype (just remove /proc) and call parent.

Reply:
I think this change is actually just not needed, dennok's pr showed you just need to modify the ship itself to work